feat(CL): twap integration (#4831)
osmosis-labs/osmosis: x/twap/README.md

This could potentially leave the store with only one record - or no records at all within the "keep" period, so the pruning mechanism keeps the newest record that is older than the pruning time. This record is necessary to enable us interpolating from and getting TWAPs from the "keep" period.
Such record is preserved for each pool.

## New Pool Types

> Post-TWAP launch, new concentrated liquidity pool types were introduced. One of such examples
> is a concentraetd liquidity pool. In the context of `x/twap`, there are subtle

Reviewer: ```suggestion
Post-TWAP launch, new pool types were introduced, one such example
being the concentrated liquidity pool. In the context of `x/twap`, there are subtle
```
